### PR TITLE
Update package and module names 

### PR DESCRIPTION
**Pull request by FloLey**
- make sure - and _ are converted correctly in the module name: the Python package name should use - while all other names (even the Python module name and directory) should use _.

- make sure capital letters are dealt with:

   - a capital letter at the start of a module name should be converted to a lower case letter
   - capital letters in other positions of the module name should not be converted

issue #14

### CODE REVIEW

File: hooks/pre_gen_project.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+"""
+    This script has as purpose to create the Python package name and the 
+    'Inmanta module name' using the module name input (or the default module name).
+    The Python package name uses the same naming convention as standard Python packages: 
+    all lowercase and words separated by "-".
+    The 'Inmanta module name' (which is also the name of the Python module name) uses "_" to separate
+    words and can have uppercase letters except for the first letter which is always lowercase.
+"""
+
+{{ cookiecutter.update({"package_name": "inmanta-module-" + cookiecutter.module_name.lower().replace("_","-") }) }}
```
Comment (sanderr) [inline hooks/pre_gen_project.py:10]: `package_name` is not relevant for v1.

**Review by sanderr — CHANGES_REQUESTED** [review]
- `module.yml` should use the inmanta module name.
- I believe for v1 the module directory should be the inmanta module name as well. This one is my bad, I only realized it now. The reason is that for v1 the compiler uses directory-based module discovery so we would break the module by using a different naming scheme.

**Comment by FloLey** [comment]
@sanderr 
I feel like the `pre_gen_project` script is not yet run when the root directory is created and therefore I can't use `{{cookiecutter.inmanta_module_name}}` as name for the module directory. When I try I have the following error:
```
Unable to create project directory '{{cookiecutter.inmanta_module_name}}'
Error message: 'collections.OrderedDict object' has no attribute 'inmanta_module_name'
Context: {
    "cookiecutter": {
        "_template": "/home/florent/Desktop/inmanta-module-template",
        "author": "Inmanta",
        "author_email": "code@inmanta.com",
        "copyright": "2022 Inmanta",
        "license": "ASL 2.0",
        "minimal_compiler_version": "2019.3",
        "module_description": "",
        "module_name": "test_module"
    }
}
```
Therefore I put pack the module_name.

**Review by arnaudsjs — APPROVED** [review]

**Comment by arnaudsjs** [comment]
> @sanderr I feel like the `pre_gen_project` script is not yet run when the root directory is created and therefore I can't use `{{cookiecutter.inmanta_module_name}}` as name for the module directory. When I try I have the following error:
> 
> ```
> Unable to create project directory '{{cookiecutter.inmanta_module_name}}'
> Error message: 'collections.OrderedDict object' has no attribute 'inmanta_module_name'
> Context: {
>     "cookiecutter": {
>         "_template": "/home/florent/Desktop/inmanta-module-template",
>         "author": "Inmanta",
>         "author_email": "code@inmanta.com",
>         "copyright": "2022 Inmanta",
>         "license": "ASL 2.0",
>         "minimal_compiler_version": "2019.3",
>         "module_description": "",
>         "module_name": "test_module"
>     }
> }
> ```
> 
> Therefore I put back the module_name.

Is this solved when we use a pre-hook instead of a post-hook. Is there a specific reason why we use a post-hook? (I am not familiar with these hooks)

**Comment by sanderr** [comment]
It's unfortunate but indeed, cookiecutter seems to [render the directory](https://github.com/cookiecutter/cookiecutter/blob/0b406256d2c0aa728f860692d57f49c0d2beb39d/cookiecutter/generate.py#L291) before [calling the pre hook](https://github.com/cookiecutter/cookiecutter/blob/0b406256d2c0aa728f860692d57f49c0d2beb39d/cookiecutter/generate.py#L313). I might create an issue on their repo but for now I guess we'll just keep this as is.

**Review by sanderr — APPROVED** [review]

**Comment by FloLey** [comment]
@arnaudsjs 
The pre_hook is used to populate the context with some additional data based on user inputs (python script)
The post_hook is used to init git and create the first commit. (bash script)

**Pull request merged**